### ISSUE TITLE
[excon 0.62.0] insecure cipher suites

### ISSUE THREAD

**Issue by nhiquach** [excon/excon#695]
Was testing Excon 0.62.0 against 3des.badssl.com and received back a 200 when I expected a 503.

Then saw this closed issue https://github.com/excon/excon/issues/688. Duplicating the command in the issue here's what I received back as output.

ruby 2.4.5
```
JSON.parse(Excon.get("https://www.howsmyssl.com/a/check").body)
=> {"given_cipher_suites"=>
  ["TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
   "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
   "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
   "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
   "TLS_DHE_RSA_WITH_AES_128_GCM_SHA256",
   "TLS_DHE_RSA_WITH_AES_256_GCM_SHA384",
   "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256",
   "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256",
   "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA",
   "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384",
   "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
   "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384",
   "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA",
   "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
   "TLS_DHE_RSA_WITH_AES_128_CBC_SHA256",
   "TLS_DHE_RSA_WITH_AES_128_CBC_SHA",
   "TLS_DHE_RSA_WITH_AES_256_CBC_SHA256",
   "TLS_DHE_RSA_WITH_AES_256_CBC_SHA",
   "TLS_ECDHE_ECDSA_WITH_3DES_EDE_CBC_SHA",
   "TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA",
   "TLS_DHE_RSA_WITH_3DES_EDE_CBC_SHA",
   "TLS_RSA_WITH_AES_128_GCM_SHA256",
   "TLS_RSA_WITH_AES_256_GCM_SHA384",
   "TLS_RSA_WITH_AES_128_CBC_SHA256",
   "TLS_RSA_WITH_AES_256_CBC_SHA256",
   "TLS_RSA_WITH_AES_128_CBC_SHA",
   "TLS_RSA_WITH_AES_256_CBC_SHA",
   "TLS_RSA_WITH_3DES_EDE_CBC_SHA",
   "TLS_EMPTY_RENEGOTIATION_INFO_SCSV"],
 "ephemeral_keys_supported"=>true,
 "session_ticket_supported"=>true,
 "tls_compression_supported"=>false,
 "unknown_cipher_suite_supported"=>false,
 "beast_vuln"=>false,
 "able_to_detect_n_minus_one_splitting"=>false,
 "insecure_cipher_suites"=>
  {"TLS_DHE_RSA_WITH_3DES_EDE_CBC_SHA"=>
    ["uses 3DES which is vulnerable to the Sweet32 attack but was not configured as a fallback in the ciphersuite order"],
   "TLS_ECDHE_ECDSA_WITH_3DES_EDE_CBC_SHA"=>
    ["uses 3DES which is vulnerable to the Sweet32 attack but was not configured as a fallback in the ciphersuite order"],
   "TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA"=>
    ["uses 3DES which is vulnerable to the Sweet32 attack but was not configured as a fallback in the ciphersuite order"]},
 "tls_version"=>"TLS 1.2",
 "rating"=>"Bad"}
```

**Comment by nhiquach** [comment]
https://github.com/excon/excon/pull/696 < my attempt at a fix

**Comment by geemus** [comment]
@nhiquach Thanks for pointing this out. I made a slightly more conservative change via: https://github.com/excon/excon/commit/ec42283e50cfd977d94651e3ca954fe0005570bd, which (on my machine/configuration at least) seems to once again satisfy howsmyssl.

It doesn't look like it causes 503s for badssl though, presumably because it still allows des as a last resort? I guess I feel more confident about sticking to a reordered version of the mozilla intermediate list to deprioritize des, than removing it altogether (as I worry this might cause backwards compatibility issues for some). Does that make sense? What do you think, is that change sufficient for your needs?

**Comment by nhiquach** [comment]
makes sense, thanks for addressing for quickly!

**Comment by geemus** [comment]
Thanks for understanding, I've released v0.63.0 including this fix.
